src/block/block_write.c

```diff
@@ -235,7 +235,7 @@ __wt_block_write(WT_SESSION_IMPL *session, WT_BLOCK *block,
 int
 __wt_block_write_off(WT_SESSION_IMPL *session, WT_BLOCK *block,
     WT_ITEM *buf, wt_off_t *offsetp, uint32_t *sizep, uint32_t *cksump,
-    bool data_cksum, bool caller_locked)
+    bool data_cksum, bool checkpoint_io, bool caller_locked)
```

Comment:
I'm sympathetic, but:

* two of the booleans (`data_cksum` & `checkpoint_io`) are shared by the btree layer, and one (`caller_locked`) is private to the block manager layer,
* the block-manager's checkpoint routine takes one of the public booleans (`data_cksum`), but not both of them,
* if there's ever another block manager, any flags would have to be shared between block managers.

We could change the block-manager's checkpoint API take `checkpoint_io` too, but we already know any I/O done for that API is checkpoint related (and the other arguments to the block-manager's checkpoint  API aren't common with the block-manager's write API).

It's a mess, but it's only a mess inside the block manager, so I went with the booleans.

Let's see what @agorrod thinks.